Analysis/Tasks/PWGHF/HFTrackIndexSkimsCreator.cxx

```diff
@@ -142,6 +173,228 @@ struct SelectTracks {
+            negTrack.pt() < ptmin) {
+          continue;
+        }
+        if (abs(posTrack.eta()) > etamax || // to the filters? I can't for now, it is not in the tables
+            abs(negTrack.eta()) > etamax) {
+          continue;
+        }
+
+        // V0 invariant mass selection
+        if (std::abs(v0.mK0Short() - massK0s) > cutInvMassV0)
+          continue; // should go to the filter, but since it is a dynamic column, I cannot use it there
+
+        // V0 cosPointingAngle selection
+        if (v0.v0cosPA(collision.posX(), collision.posY(), collision.posZ()) < cospaV0)
+          continue;
+
+        auto posTrackParCov = getTrackParCov(posTrack);
+        posTrackParCov.propagateTo(v0.posX(), d_bz); // propagate the track to the X closest to the V0 vertex
+        auto negTrackParCov = getTrackParCov(negTrack);
+        negTrackParCov.propagateTo(v0.negX(), d_bz); // propagate the track to the X closest to the V0 vertex
+        std::array<float, 3> pvecv0 = {0., 0., 0.};
+        std::array<float, 3> pvecbach = {0., 0., 0.};
+
+        const std::array<float, 3> vertexV0 = {v0.x(), v0.y(), v0.z()};
+        const std::array<float, 3> momentumV0 = {v0.px(), v0.py(), v0.pz()};
+
+        // we build the neutral track to then build the cascade
+        auto trackV0 = o2::dataformats::V0(vertexV0, momentumV0, posTrackParCov, negTrackParCov, {0, 0}, {0, 0}); // build the V0 track
+
+        // now we find the DCA between the V0 and the bachelor, for the cascade
+        int nCand2 = fitter.process(trackV0, bachTrack);
+
+        std::array<float, 3> pVecCandCasc = {0., 0., 0.};
+        if (nCand2 != 0) {
+          fitter.propagateTracksToVertex();        // propagate the bach and V0 to the Lc vertex
+          fitter.getTrack(0).getPxPyPzGlo(pvecv0); // take the momentum at the Lc vertex
+          fitter.getTrack(1).getPxPyPzGlo(pvecbach);
+
+          // HF code for the 2 prongs would use:
+          //auto pVecCandCasc = array{bachTrack.px() + trackV0.px(),
+          //			       bachTrack.py() + trackV0.py(),
+          //			       bachTrack.pz() + trackV0.pz()};
+          // but I think this is equivalent, and I understand it better where it comes from
+          pVecCandCasc = array{pvecbach[0] + pvecv0[0],
+                               pvecbach[1] + pvecv0[1],
+                               pvecbach[2] + pvecv0[2]};
+
+          // cascade candidate pT cut
+          if (RecoDecay::Pt(pVecCandCasc) < cutCascPtCandMin) {
+            continue;
+          }
+
+          // invariant mass
+          // calculate invariant masses
+          auto arrMom = array{pvecbach, pvecv0};
+          mass2K0sP = RecoDecay::M(arrMom, array{massP, massK0s});
+          // invariant-mass cut
+          if (cutCascInvMassLc >= 0.) {
+            if (std::abs(mass2K0sP - massLc) < cutCascInvMassLc) {
+              continue;
+            }
+          }
```

Comment:
The two `if`s can be merged.
For 2-prong skims, the mass cut is applied before vertexing. Is there a reason to do it after in this case?

Reply:
I don't understand: how can I do it before? I need the momenta at the secondary vertex. Maybe I am overlooking something...

Reply:
BTW, now there is 1 if. Not sure it has any impact though...